created signup page

mukundmadhav054 opened the pull request.

mukundmadhav054 commented:
@CodiumAI-Agent /review

CodiumAI-Agent commented:
## PR Review

<table>
<tr>
<tr><td> ⏱️&nbsp;<strong>Estimated&nbsp;effort&nbsp;to&nbsp;review [1-5]</strong></td><td>

3, because the PR introduces several new components and significant changes to the routing and layout of the application. The complexity of the changes and the need to ensure that the new routing and components integrate seamlessly with the existing application architecture require a careful review.


</td></tr>
<tr><td> 🧪&nbsp;<strong>Relevant tests</strong></td><td>

No


</td></tr>
<tr><td rowspan=2> 🔍&nbsp;<strong>Possible issues</strong></td>
<td>

<strong>Possible Bug:</strong> The `useLocation` hook is used in `App.jsx` without wrapping the component tree in a `<Router>` component from `react-router-dom`. This might cause the application to crash or not function as expected when trying to access `location.pathname`.</td></tr>
<tr>
<td>

<strong>Performance Concern:</strong> In `Navbar.jsx`, the `handleResize` function is directly attached to the window resize event, which can lead to performance issues on frequent resizing. Debouncing the resize event handler is recommended to mitigate potential performance degradation.</td></tr>
<tr><td> 🔒&nbsp;<strong>Security concerns</strong></td><td>

No

</td></tr>
</table>


<details><summary> <strong>Code feedback:</strong></summary>

<hr><table><tr><td>relevant file</td><td>client/src/App.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Ensure that the root component (`<App />`) is wrapped in a `<Router>` component from `react-router-dom` to properly use the `useLocation` hook. Without wrapping the application in a `<Router>`, the `useLocation` hook will not function correctly, potentially leading to runtime errors. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S54_MukundMadhav_Capstone_HabTrackIt/pull/16/files#diff-51b002bf85fbd146b38871b89ba0d41c2bead37756295748c67206aac23ddc9aR10'>const location = useLocation();</a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>client/src/components/Navbar.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Consider debouncing the `handleResize` function attached to the window's resize event. This can improve performance by limiting the rate at which the function is executed during window resizing. You can use lodash's `debounce` function or implement your own debouncing logic. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S54_MukundMadhav_Capstone_HabTrackIt/pull/16/files#diff-5d730edff1e62dd349f6963f21c7b70440e9f084186ffc03d533926ca8768886R26'>window.addEventListener("resize", handleResize);</a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>client/src/components/Signup.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Add `noopener noreferrer` to the `rel` attribute of external links to enhance security. This prevents the newly opened page from being able to access the `window.opener` property and ensures the browser does not pass the referrer header to the new page. [important]

</strong>
</td></tr><tr><td>relevant line</td><td><a href='https://github.com/kalviumcommunity/S54_MukundMadhav_Capstone_HabTrackIt/pull/16/files#diff-650c27a3769b3132287e73e8b28015c4c3d759dc65bd38c018d64ec1a099ad24R30'><Image src="./LinkedIn.svg" /></a></td></tr></table><hr>

<table><tr><td>relevant file</td><td>client/src/components/SignupForm.jsx
</td></tr><tr><td>suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td>

<strong>

Consider using a more descriptive `alt` attribute for the `Image` components for better accessibility. Providing a meaningful description helps users who rely on screen readers to understand the content of the images. [medium]

</strong>
</td></tr><tr><td>relevant line</td><td><Image src="./LinkedIn.svg" /></td></tr></table><hr>

</details><hr>

<details> <summary><strong>✨ Review tool usage guide:</strong></summary><hr> 

**Overview:**
The `review` tool scans the PR code changes, and generates a PR review which includes several types of feedbacks, such as possible PR issues, security threats and relevant test in the PR. More feedbacks can be [added](https://pr-agent-docs.codium.ai/tools/review/#general-configurations) by configuring the tool.

The tool can be triggered [automatically](https://pr-agent-docs.codium.ai/usage-guide/automations_and_usage/#github-app-automatic-tools-when-a-new-pr-is-opened) every time a new PR is opened, or can be invoked manually by commenting on any PR.
- When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L23) related to the review tool (`pr_reviewer` section), use the following template:
```
/review --pr_reviewer.some_config1=... --pr_reviewer.some_config2=...
```
- With a [configuration file](https://pr-agent-docs.codium.ai/usage-guide/configuration_options/), use the following template:
```
[pr_reviewer]
some_config1=...
some_config2=...
```
    

See the review [usage page](https://pr-agent-docs.codium.ai/tools/review/) for a comprehensive guide on using this tool.


</details>

mukundmadhav054 commented:
1. I was using browser router in the main.jsx file so it doesn't support router inside router. (Irrelevant)
2. Added lodash and used debounce in the resize function.
3. Irrelevant.
4. Added alt for all the social link logos.
